docs/tutorials/byproduct_recovery_determination.ipynb

```diff
@@ -0,0 +1,706 @@
+    "__1. Specify Product Prices__\n",
+    "\n",
+    "Product prices are stored in ``costing_dictionary.py``. You can access the default sale prices using the function ``load_default_sale_prices()`` :"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "dd13f5b4-1ceb-4755-a220-950708a60bff",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sale_prices = load_default_sale_prices()\n",
+    "Li_price = sale_prices[\"Li\"]\n",
+    "Co_price = sale_prices[\"Co\"]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e82adab5-7e30-448c-b7ca-f0000e2265e6",
+   "metadata": {},
+   "source": [
+    "These prices will be used to compute __annual revenue__, which is a key input to the byproduct recovery decision framework.\n",
+    "\n",
+    "__2. Specify the Product Putputs__\n",
+    "\n",
+    "Next, define the __products__ recovered from the byproduct recovery plant.\n",
+    "\n",
+    "The __byproduct recovery framework__ accepts flexible inputs, where products are listed in a structure called ``material_list``, and their associated data (e.g. flow rate, price) are organized in a dictionary called ``material_data``.\n",
+    "- ``material_list``: A list of product names (strings).\n",
+    "- ``material_data``: A dictionary where each key corresponds to a product name and maps to its properties such as flow rate and price.\n",
+    "\n",
+    "This structured input will be passed into the byproduct recovery decision framework to assess economic viability."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "a99db64a-b1b2-4287-a46b-b20d58149143",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The framework take flexible material input\n",
+    "material_list = [\"Lithium\", \"Cobalt\"]\n",
+    "\n",
+    "# Introduce the byproduct recovery framework\n",
+    "model = ConcreteModel()\n",
+    "model.recovery_determine = ByproductRecovery(materials=material_list)\n",
+    "\n",
+    "# Define input values based on provided materials\n",
+    "material_data = {\n",
+    "    \"Lithium\": {\n",
+    "        \"production\": Li_recovery_mass,\n",
+    "        \"market_value\": Li_price,\n",
+    "        \"waste_disposal\": 1,\n",
+    "        \"conversion\": 0,\n",
+    "        \"conversion_cost\": 0,\n",
+    "        \"process_steps\": 1,\n",
+    "        \"process_cost\": total_annualized_cost,\n",
+    "    },\n",
+    "    \"Cobalt\": {\n",
+    "        \"production\": Co_recovery_mass,\n",
+    "        \"market_value\": Co_price,\n",
+    "        \"waste_disposal\": 1,\n",
+    "        \"conversion\": 0,\n",
+    "        \"conversion_cost\": 0,\n",
+    "        \"process_steps\": 0,\n",
+    "        \"process_cost\": 0,\n",
+    "    },\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dbe5705c-f654-4190-b490-87cdc41579dd",
+   "metadata": {},
+   "source": [
+    "__3. Set Additional Parameters and Evaluate Net Benefit__\n",
```

Comment:
I suggest only putting backticks around text if it's code object names, otherwise use bold or italics instead.